OpenRA.Mods.Common/Traits/Render/WithChargeOverlay.cs

```diff
@@ -0,0 +1,62 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Rendered together with AttackCharge.")]
+	public class WithChargeOverlayInfo : ITraitInfo, Requires<RenderSpritesInfo>
+	{
+		[Desc("Sequence name to use")]
+		public readonly string Sequence = "active";
+
+		[Desc("Custom palette name")]
+		public readonly string Palette = null;
+
+		[Desc("Custom palette is a player palette BaseName")]
+		public readonly bool IsPlayerPalette = false;
+
+		public object Create(ActorInitializer init) { return new WithChargeOverlay(init, this); }
+	}
+
+	public class WithChargeOverlay : INotifyCharging, INotifyDamageStateChanged
+	{
+		readonly Animation overlay;
+		readonly RenderSprites renderSprites;
+		readonly WithChargeOverlayInfo info;
+
+		bool charging;
+
+		public WithChargeOverlay(ActorInitializer init, WithChargeOverlayInfo info)
+		{
+			this.info = info;
+
+			renderSprites = init.Self.Trait<RenderSprites>();
+
+			overlay = new Animation(init.World, renderSprites.GetImage(init.Self));
```

Comment:
You need a NormalizeSequence here too, otherwise it will display the wrong sequence if it is spawned damaged.